sui_core/src/authority_active/gossip/mod.rs

```diff
@@ -86,6 +86,7 @@ where
                 // Are we likely to never terminate because of this condition?
                 // - We check we have nodes left by stake
                 // - We check that we have at least 2/3 of nodes that can be contacted.
+                tokio::time::sleep(Duration::from_millis(10)).await;
```

Comment:
is 10ms enough?

Reply:
> In the context of checkpoints, when an authority sends a message through consensus, does it need wait for the message to be sequenced to then perform further processing? If so, where does this happen? It seems we are not waiting for a response:

Nope, the assumption is that the checkpoint function `handle_internal_fragment` will be called when a fragment is sequenced. No need to "wait" for it.

Reply:
I see, so you mean that it is the sender's job to check after some time and retry?

Reply:
yes, we should already have the retry logic implemented in the client (needed by traditional users to submit their shared-object txs)